src/logic/wallets/getWeb3.ts

```diff
@@ -104,3 +106,10 @@ export const isTxPendingError = (err: Error): boolean => {
   const WEB3_TX_NOT_MINED_ERROR = 'Transaction was not mined within'
   return (err.message || '').startsWith(WEB3_TX_NOT_MINED_ERROR)
 }
+
+export const getSDKWeb3Adapter = async (): Promise<Web3Adapter> => {
+  const web3 = getWeb3()
+  const signerAddress = (await getAccountFrom(web3)) || ZERO_ADDRESS
```

Comment:
This could be made synchronous if we grab the account address from Redux, wdyt?
<img width="483" alt="Screenshot 2021-09-22 at 11 34 46" src="https://user-images.githubusercontent.com/381895/134319822-8df9c2b0-c105-41ae-b6c8-1eb1c14619fc.png">

Reply:
You could also take signerAddress as a param, the function will be way more flexible this way

Reply:
What would be the use case, @mikheevm?

Reply:
You can use the function and define how you get an address (redux, async, sync, zero address) - it doesn't matter for the function—FP style.

Reply:
The idea for this particular function is to make it super easy to call the SDK from elsewhere in the code. And to reduce the boilerplate. So unless there's a business logic use case to pass a _different_ signer address than the current one, I wouldn't add it.

Reply:
There's business logic built in this function with the zero address

Reply:
I've updated it to have the current account (taken from the store) as the default value. Estimating with it, it's now passed `ZERO_ADDRESS`.